Common/Securities/SecurityDefinitionSymbolResolver.cs

```diff
@@ -0,0 +1,198 @@
+        /// </param>
+        /// <param name="tradingDate">
+        /// The date that the stock was trading at with the SEDOL provided. This is used
+        /// to get the ticker of the symbol on this date.
+        /// </param>
+        /// <returns>The Lean Symbol corresponding to the SEDOL on the trading date provided</returns>
+        public Symbol SEDOL(string sedol, DateTime tradingDate)
+        {
+            if (string.IsNullOrWhiteSpace(sedol))
+            {
+                return null;
+            }
+            
+            return SecurityDefinitionToSymbol(
+                _securityDefinitions?.FirstOrDefault(x => x.SEDOL == sedol),
+                tradingDate);
+        }
+
+        /// <summary>
+        /// Converts ISIN into a Lean <see cref="Symbol"/>
+        /// </summary>
+        /// <param name="isin">
+        /// The International Securities Identification Number (ISIN) of a security
+        /// </param>
+        /// <param name="tradingDate">
+        /// The date that the stock was trading at with the ISIN provided. This is used
+        /// to get the ticker of the symbol on this date.
+        /// </param>
+        /// <returns>The Lean Symbol corresponding to the ISIN on the trading date provided</returns>
+        public Symbol ISIN(string isin, DateTime tradingDate)
+        {
+            if (string.IsNullOrWhiteSpace(isin))
+            {
+                return null;
+            }
+            
+            return SecurityDefinitionToSymbol(
+                _securityDefinitions?.FirstOrDefault(x => x.ISIN == isin),
+                tradingDate);
+        }
+
+        /// <summary>
+        /// Converts a SecurityDefinition to a <see cref="Symbol" />
+        /// </summary>
+        /// <param name="securityDefinition">Security definition</param>
+        /// <param name="tradingDate">
+        /// The date that the stock was being traded. This is used to resolve
+        /// the ticker that the stock was trading under on this date.
+        /// </param>
+        /// <returns>Symbol if matching Lean Symbol was found on the trading date, null otherwise</returns>
+        private Symbol SecurityDefinitionToSymbol(SecurityDefinition securityDefinition, DateTime tradingDate)
+        {
+            if (securityDefinition == null)
+            {
+                return null;
+            }
+
+            var market = securityDefinition.SecurityIdentifier.Market;
+            if (!_mapFileResolvers.TryGetValue(market, out var mapFileResolver))
+            {
+                var mapFileProvider = Composer.Instance.GetExportedValueByTypeName<IMapFileProvider>(Config.Get("map-file-provider", "LocalDiskMapFileProvider"));
+                mapFileProvider.Initialize(_dataProvider);
+
+                mapFileResolver = mapFileProvider.Get(market);
+                _mapFileResolvers[market] = mapFileResolver;
+            }
```

Comment:
Think the initial approach of performing the fetching and init of the `mapFileProvider` on construction seems more accurate since it should only be done once 👍 
We don't really need a cache of the `mapFileResolver`,  the provider handles it internally, and the provider handles clearing and regenerating it's cache which wouldn't happen here 👍 - Let's just do `var mapFileResolver = mapFileProvider.Get(market);`